### PR TITLE
Pass config separately from filters and make it extendable

### PR DESCRIPTION
**Pull request by axelboc**

### CODE REVIEW

File: src/App/adapter/lib/create.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ const byTargetLength = (list) => {
 const buildGroup = (group, acc) => {
   const { filters, operator, target } = group
 
-  const relation = target.at(-1)
+  const relation = target[target.length - 1]
```
Comment (axelboc) [inline src/App/adapter/lib/create.js:34]: A bit too recent for Jest, it seems.

File: src/App/adapter/translate.js [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
+
+const DEFAULT_CONFIG = {
+  endpoint: 'documents',
+  include: [],
+  pageSize: 5, // `false` to disable limit/pagination
+  page: 1,
+}
+
+function translate(diffState, initialState = [], queryConfig = {}) {
+  const config = { ...DEFAULT_CONFIG, ...queryConfig }
```
Comment (axelboc) [inline src/App/adapter/translate.js:13]: Merging the custom query config with the default config. See examples in tests.

I'll put the `initialFilters` in the config object in my next PR.

File: src/App/adapter/lib/state.js [hunk 3]
```diff
@@ -22,37 +22,27 @@ const resolvePath = (label, endpoint) => {
...
 
-  const {
-    label: _,
-    name: __,
-    value: ___,
-    include: included = [],
-    ...base
-  } = enhancePagination(config)
+export function parseState(state, config) {
+  const { endpoint, include: included } = config
```
Comment (axelboc) [inline src/App/adapter/lib/state.js:37]: `parseState` no longer has to look for the config object and worry about returning a `base` object -- it receives the config as argument. The pagination object is created in `createPagination`, which is now called directly from `translate`.

**Comment by noobadmin** [comment]
Hmmm, wouldn't at least "order" be useful if settable via uri?

**Comment by axelboc** [comment]
Sure, well spotted! Are there other fields that need to be forwarded? Better to be explicit about it, so we don't end up with invalid properties in the query.

**Comment by axelboc** [comment]
Doesn't seem like there is, so I'm going to just forward `order`.

**Comment by noobadmin** [comment]
I don't think so. At least it should be getting known keys instead of dumpung wrong ones

File: src/App/adapter/translate.js [hunk 4]
```diff
@@ -1,18 +1,30 @@
...
 import { stripEmptyKeys } from './lib/helpers'
-import { mergeState, parseState } from './lib/state'
+import { createPagination, mergeState, parseState } from './lib/state'
+
+const DEFAULT_CONFIG = {
+  endpoint: 'documents',
+  include: [],
+  pageSize: 5, // `false` to disable limit/pagination
+  page: 1,
+  order: undefined, // e.g.`['foo ASC', 'bar DESC']`
```
Comment (axelboc) [inline src/App/adapter/translate.js:10]: Declaring `order` explicitly here for documentation purposes. If we ever switch to TypeScript, this won't be needed.

**Comment by axelboc** [comment]
I'm going to add a test to check the forwarding.

File: src/App/adapter/translate.js (hunk 2 above)
Comment (noobadmin) [reply src/App/adapter/translate.js:14]: It feels more elegant to me to achieve this via mergeState

File: src/App/adapter/lib/state.js (hunk 3 above)
Comment (noobadmin) [reply src/App/adapter/lib/state.js:37]: Not seeing the cost here.

File: src/App/adapter/translate.js (hunk 2 above)
Comment (noobadmin) [reply src/App/adapter/translate.js:14]: Your decision though! Mine have a history of turning out "a bit silly".

**Comment by noobadmin** [comment]
Fields & include might actually come in handy as well, depends on the implementation of the detail page.

File: src/App/adapter/lib/state.js (hunk 3 above)
Comment (axelboc) [reply src/App/adapter/lib/state.js:37]: It's just less convoluted, in my opinion. `mergeState` has enough to deal with as it is. Generating the `include` and `where` objects should be fully isolated from dealing with the "global" query configuration (`order`, `limit` and `skip`).

**Comment by axelboc** [comment]
> Fields & include might actually come in handy as well, depends on the implementation of the detail page.

Sorry, can you please elaborate?

File: src/App/adapter/translate.js (hunk 2 above)
Comment (axelboc) [reply src/App/adapter/translate.js:14]: No worries, I'm hoping that my reasoning will make more sense once I show you some code.

**Pull request merged**